youtube_dl/extractor/kontrtube.py

```diff
@@ -31,16 +31,16 @@ class KontrTubeIE(InfoExtractor):
 
     def _real_extract(self, url):
         mobj = re.match(self._VALID_URL, url)
-        video_id = mobj.group('id')
-        display_id = mobj.group('display_id')
+        video_id, display_id = mobj.groups()
 
-        webpage = self._download_webpage(
-            url, display_id, 'Downloading page')
+        webpage = self._download_webpage(url, display_id)
+
+        video_data = self._parse_json(self._search_regex(
+            r'var\s+flashvars\s*=\s*({[^}]+});', webpage, 'video data'),
+        video_id, transform_source=js_to_json)
```

Comment:
I've already pointed out: replacing one time-proven working data extraction approach with another is not an improvement and I will reject all such further PRs. All additional scenarios should be added as fallbacks.
Again, this code is not PEP 8 compliant.